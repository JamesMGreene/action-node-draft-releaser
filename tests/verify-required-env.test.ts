import path from 'path';
import verifyRequiredEnv from '../src/verify-required-env';

describe('verifyRequiredEnv', () => {
  afterEach(() => {
    // Environment variables
    delete process.env.GITHUB_EVENT_NAME;
    delete process.env.GITHUB_REF;
    delete process.env.GITHUB_REPOSITORY;
    delete process.env.GITHUB_SHA;
    delete process.env.GITHUB_WORKSPACE;

    // Secrets
    delete process.env.GITHUB_TOKEN;
  });

  it('should not throw if all required environment variables and secrets have values', () => {
    // Environment variables
    process.env.GITHUB_EVENT_NAME = 'push';
    process.env.GITHUB_REF = 'refs/heads/master';
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = 'deadbeefcafe1c6e148a9be56893bed50abadbed';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    // Secrets
    process.env.GITHUB_TOKEN = 'testToken123';

    expect(verifyRequiredEnv()).toBe(undefined);
  });

  it('should throw if none of the required environment variables or secrets exist', () => {
    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if none of the required environment variables exist', () => {
    // Secrets
    process.env.GITHUB_TOKEN = 'testToken123';

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if none of the required secrets exist', () => {
    // Environment variables
    process.env.GITHUB_EVENT_NAME = 'push';
    process.env.GITHUB_REF = 'refs/heads/master';
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = 'deadbeefcafe1c6e148a9be56893bed50abadbed';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if all of the required environment variables and secrets are missing values', () => {
    // Environment variables
    process.env.GITHUB_EVENT_NAME = '';
    process.env.GITHUB_REF = '';
    process.env.GITHUB_REPOSITORY = '';
    process.env.GITHUB_SHA = '';
    process.env.GITHUB_WORKSPACE = '';

    // Secrets
    process.env.GITHUB_TOKEN = '';

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if all of the required environment variables are missing values', () => {
    // Environment variables
    process.env.GITHUB_EVENT_NAME = '';
    process.env.GITHUB_REF = '';
    process.env.GITHUB_REPOSITORY = '';
    process.env.GITHUB_SHA = '';
    process.env.GITHUB_WORKSPACE = '';

    // Secrets
    process.env.GITHUB_TOKEN = 'testToken123';

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if all of the required secrets are missing values', () => {
    // Environment variables
    process.env.GITHUB_EVENT_NAME = 'push';
    process.env.GITHUB_REF = 'refs/heads/master';
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = 'deadbeefcafe1c6e148a9be56893bed50abadbed';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    // Secrets
    process.env.GITHUB_TOKEN = '';

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if some of the required environment variables and secrets are missing values', () => {
    // Environment variables
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = '';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    // Secrets
    process.env.GITHUB_TOKEN = '';

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if some of the required environment variables are missing values', () => {
    // Environment variables
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = '';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    // Secrets
    process.env.GITHUB_TOKEN = 'testToken123';

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });

  it('should throw if some of the required environment variables and secrets are missing values', () => {
    // Environment variables
    process.env.GITHUB_EVENT_NAME = 'push';
    process.env.GITHUB_REF = 'refs/heads/master';
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = 'deadbeefcafe1c6e148a9be56893bed50abadbed';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    // Secrets
    process.env.GITHUB_TOKEN = '';

    expect(verifyRequiredEnv).toThrowErrorMatchingSnapshot();
  });
});
