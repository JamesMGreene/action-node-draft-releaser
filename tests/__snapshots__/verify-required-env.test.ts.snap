// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verifyRequiredEnv should throw if all of the required environment variables and secrets are missing values 1`] = `
"The following environment variables are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_REPOSITORY
- GITHUB_SHA
- GITHUB_WORKSPACE"
`;

exports[`verifyRequiredEnv should throw if all of the required environment variables are missing values 1`] = `
"The following environment variables are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_REPOSITORY
- GITHUB_SHA
- GITHUB_WORKSPACE"
`;

exports[`verifyRequiredEnv should throw if all of the required secrets are missing values 1`] = `
"The following secrets are required for this GitHub Action to run:
- GITHUB_TOKEN"
`;

exports[`verifyRequiredEnv should throw if none of the required environment variables exist 1`] = `
"The following environment variables are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_REPOSITORY
- GITHUB_SHA
- GITHUB_WORKSPACE"
`;

exports[`verifyRequiredEnv should throw if none of the required environment variables or secrets exist 1`] = `
"The following environment variables are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_REPOSITORY
- GITHUB_SHA
- GITHUB_WORKSPACE"
`;

exports[`verifyRequiredEnv should throw if none of the required secrets exist 1`] = `
"The following secrets are required for this GitHub Action to run:
- GITHUB_TOKEN"
`;

exports[`verifyRequiredEnv should throw if some of the required environment variables and secrets are missing values 1`] = `
"The following environment variables are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_SHA"
`;

exports[`verifyRequiredEnv should throw if some of the required environment variables and secrets are missing values 2`] = `
"The following secrets are required for this GitHub Action to run:
- GITHUB_TOKEN"
`;

exports[`verifyRequiredEnv should throw if some of the required environment variables are missing values 1`] = `
"The following environment variables are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_SHA"
`;
